### ISSUE TITLE
Драйвер и потоки

### ISSUE THREAD

**Issue by capcom6** [evotor/device-drivers#22]
Здравствуйте.

Поясните, пожалуйста, если я не ошибаюсь, сервис в Android выполняется по-умолчанию в потоке UI, отсюда следует, что драйвер должен выполнять взаимодействие с оборудованием в отдельном потоке.
Однако, в случае драйвера принтера этикеток, должен ли драйвер блокировать работу в методе printPrice пока этикетка не распечатана? Или можно делать это в отдельном потоке и возвращаться из метода до конца печати?

**Comment by rMozg** [comment]
Да, следует блокировать работу в методе printPrice до тех пор, пока этикетка не будет успешно распечатана. Если вернуться из метода раньше этого, то не получится сообщить вызывающему приложению о какой-либо возникшей ошибке.
